test/bin/scenario.sh

```diff
@@ -497,6 +497,61 @@ remove_vm() {
     rm -rf "${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${vmname}"
 }
 
+# Configure the firewall in the VM based on the instructions in the documentation.
+configure_vm_firewall() {
+    local -r vmname="$1"
+
+    local -r api_port=$(get_vm_property "${vmname}" api_port)
+
+    # ssh, just to be sure
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=public --add-port=22/tcp"
+
+    # Installation instructions
+    # - On-host pod communication
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=trusted --add-source=10.42.0.0/16"
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=trusted --add-source=169.254.169.1"
+
+    # Networking / firewall configuration instructions
+    # - Incoming for the router
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=public --add-port=80/tcp"
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=public --add-port=443/tcp"
+    # - mdns
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=public --add-port=5353/udp"
+    # - Incoming for the API server
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=public --add-port=${api_port}/tcp"
+    # - Incoming for NodePort services
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=public --add-port=30000-32767/tcp"
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --permanent --zone=public --add-port=30000-32767/udp"
+
+    run_command_on_vm "${vmname}" "sudo firewall-cmd --reload"
+}
+
+# Enable subscription in the VM
+subscription_manager_register() {
+    local -r vmname="$1"
+
+    if [ -f /tmp/subscription-manager-org ]; then
+        # CI workflow
+        cat <<EOF > /tmp/sub.sh
+#!/bin/bash
+set -xeuo pipefail
+
+if ! sudo subscription-manager status >&/dev/null; then
+    sudo subscription-manager register \
+        --org="\$(cat /tmp/subscription-manager-org)" \
+        --activationkey="\$(cat /tmp/subscription-manager-act-key)"
+fi
+EOF
+        copy_file_to_vm "${vmname}" /tmp/sub.sh /tmp/sub.sh
+        copy_file_to_vm "${vmname}" /tmp/subscription-manager-org /tmp/subscription-manager-org
+        copy_file_to_vm "${vmname}" /tmp/subscription-manager-act-key /tmp/subscription-manager-act-key
+        run_command_on_vm "${vmname}" "chmod +x /tmp/sub.sh && sudo /tmp/sub.sh"
+    else
+        # Local developer workflow
+        run_command_on_vm "${vmname}" "sudo subscription-manager register"
+    fi
```

Comment:
I would like to suggest a few alternatives for the current implementation. The goal is not to include CI-specific code in the `scenario.sh` generic script and avoid prompt when running tests in devenv:

- Introduce a scenario setting parameter pointing to subscription information
- Move org and activationKey detection code to the scenario and pass the values to the function.
- Can we use the subscription technique used by containers?

Reply:
This commit is really part of #2568 so I made the change there. I will rebase this PR when that one lands.